amazon/ion/reader_binary.py

```diff
@@ -194,7 +195,11 @@ def _parse_decimal(buf):
         value = Decimal((sign_bit, (0,), exponent))
     else:
         coefficient *= sign_bit and -1 or 1
-        value = Decimal(coefficient).scaleb(exponent)
+        with localcontext() as context:
+            # Adjusting precision for taking into account arbitrarily
+            # large/small numbers
+            context.prec = len(str(coefficient)) + abs(exponent)
```

Comment:
Let's remove `+ abs(exponent)` from this expression.  Setting precision to the length of the coefficient should be sufficient.

Reply:
Changes made.